src/layout/layout.rs

```diff
@@ -1212,5 +1273,288 @@ mod tests {
                 ]
             );
         }
+
+        #[test]
+        fn proportional_fixed() {
```

Comment:
These are super clear and maintainable. Will read them more thoroughly later, but just wanted to call that out as really nice.